### PR TITLE
Rs/marlin downstream v0.3.2

### PR DESCRIPTION
**Pull request by robertgshaw2-neuralmagic**

### CODE REVIEW

File: tests/models/test_marlin.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        gptq_output_ids, gptq_output_str, gptq_logprobs = gptq_outputs[
+            prompt_idx]
+        marlin_output_ids, marlin_output_str, marlin_logprobs = marlin_outputs[
+            prompt_idx]
+
+        for idx, (gptq_output_id, marlin_output_id) in enumerate(
+                zip(gptq_output_ids, marlin_output_ids)):
+            # If sequence is not an exact match ...
+            if marlin_output_id != gptq_output_id:
+                # ... each predicted token must be in top 5 of the other's
```
Comment (tlrmchlsmth) [inline tests/models/test_marlin.py:88]: where does the number 5 come in?

File: tests/models/test_marlin.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    dtype: str,
+    max_tokens: int,
+    num_logprobs: int,
+) -> None:
+    marlin_model = vllm_runner(model_pair.model_marlin, dtype=dtype)
+    marlin_outputs = marlin_model.generate_greedy_logprobs(
+        example_prompts, max_tokens, num_logprobs)
+
+    # Note: not sure why, but deleting just the model on Ada Lovelace
+    #   does not free the GPU memory. On Ampere, deleting the just model
```
Comment (andy-neuma) [inline tests/models/test_marlin.py:61]: looks like extra "just"

File: tests/models/test_marlin.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    del marlin_model.model.llm_engine.driver_worker
+    del marlin_model
+
+    gptq_model = vllm_runner(model_pair.model_gptq, dtype=dtype)
+    gptq_outputs = gptq_model.generate_greedy_logprobs(example_prompts,
+                                                       max_tokens,
+                                                       num_logprobs)
+
+    # Note: not sure why, but deleting just the model on Ada Lovelace
+    #   does not free the GPU memory. On Ampere, deleting the just model
```
Comment (andy-neuma) [inline tests/models/test_marlin.py:72]: extra "just"

File: tests/models/test_marlin.py (hunk 1 above)
Comment (tlrmchlsmth) [reply tests/models/test_marlin.py:88]: oh, maybe it's actually 3, not 5, since num_logprobs 3?

File: vllm/config.py [hunk 4]
```diff
@@ -182,15 +182,20 @@ def _verify_sparsity(self) -> None:
...
         if self.quantization is not None:
             self.quantization = self.quantization.lower()
 
         # Parse quantization method from the HF model config, if available.
         hf_quant_config = getattr(self.hf_config, "quantization_config", None)
         if hf_quant_config is not None:
             hf_quant_method = str(hf_quant_config["quant_method"]).lower()
+            # If the GPTQ model is serialized in marlin format, use marlin.
+            if (hf_quant_method == "gptq"
+                    and "is_marlin_format" in hf_quant_config
```
Comment (andy-neuma) [inline vllm/config.py:196]: might be good to hoist the string "is_marlin_format" into a variable to avoid typos.

File: tests/models/test_marlin.py
Comment (tlrmchlsmth) [inline tests/models/test_marlin.py:1]: do we need to add this test to a workflow?

**Review by andy-neuma — APPROVED** [review]
cool

File: vllm/engine/arg_utils.py [hunk 5]
```diff
@@ -297,14 +297,12 @@ def create_engine_configs(
...
-                                   self.code_revision, self.tokenizer_revision,
-                                   self.max_model_len, self.quantization, 
-                                   self.sparsity, self.enforce_eager,
-                                   self.max_context_len_to_capture)
+        model_config = ModelConfig(
+            self.model, self.tokenizer, self.tokenizer_mode,
+            self.trust_remote_code, self.download_dir, self.load_format,
+            self.dtype, self.seed, self.revision, self.code_revision,
+            self.tokenizer_revision, self.max_model_len, self.quantization,
+            self.sparsity, self.enforce_eager, self.max_context_len_to_capture)
```
Comment (tlrmchlsmth) [inline vllm/engine/arg_utils.py:305]: looks like only a whitespace change here unless I'm missing something. Makes me concerned about the potential of merge conflicts due to changes like this. Are both we and the vllm folks using black?

Comment (mgoin) [reply vllm/engine/arg_utils.py:305]: `yapf==0.32.0` should be used per the `requirements.txt`

**Review by tlrmchlsmth — APPROVED** [review]

File: vllm/engine/arg_utils.py (hunk 5 above)
Comment (robertgshaw2-neuralmagic) [reply vllm/engine/arg_utils.py:305]: This was done automatically by `./format.sh`, Ill revert

File: tests/models/test_marlin.py (hunk 1 above)
Comment (robertgshaw2-neuralmagic) [reply tests/models/test_marlin.py:88]: Sorry, I originally had 5 + then switched to 3

Switched to Top N

File: tests/models/test_marlin.py
Comment (robertgshaw2-neuralmagic) [reply tests/models/test_marlin.py:1]: Andy's automation will catch this since we are going dir by dir

File: vllm/engine/arg_utils.py (hunk 5 above)
Comment (robertgshaw2-neuralmagic) [reply vllm/engine/arg_utils.py:305]: note: we fail the linting automation if I dont have this

**Review by mgoin — APPROVED** [review]

**Pull request merged**